### PR TITLE
View template or subplan list

### PR DESCRIPTION
**Pull request by JackSekoranja**
Completed Issue #37 (View Template or Subplan List (5)). 

This page is located at the address `http://127.0.0.1:8000/list`, and displays a list of templates and a list of subplans that are contained in the database. 

If you are notified that no templates/subplans exist, you need to add them (`http://127.0.0.1:8000/api/model/subplan/`, `http://127.0.0.1:8000/api/model/degree/`).

Template:Subplan List

### CODE REVIEW

File: cassdegrees/templates/list.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        {% for row in subplans %}
+            <tr>
+                {% for _, item in row.items %}
+                    <td>{{ item }}</td>
+                {% endfor %}
+            </tr>
+        {% endfor %}
+    </table>
+    {% if not subplans %}<p>No subplans to display</p>{% endif %}
+{% endblock %}
```
Comment (j-selby) [inline cassdegrees/templates/list.html]: Newline.

File: cassdegrees/templates/list.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% extends "base.html" %}
+
+{% block page-title %}Template/Subplan List{% endblock %}
+
+{% block subtitle %}Templates{% endblock %}
+
+{% block content %}
+    <table class="fullwidth tbl-cell-bdr">
```
Comment (j-selby) [inline cassdegrees/templates/list.html]: Perhaps http://webpublishing.anu.edu.au/web-style-guide/page-tabs.php would be the way to go here.

File: cassdegrees/ui/urls.py [hunk 3]
```diff
@@ -15,9 +15,10 @@
 """
 from django.contrib import admin
 from django.urls import path
-from .views import index, sampleform
+from .views import index, sampleform, planList
 
 urlpatterns = [
     path('', index),
     path('sampleform/', sampleform),
+    path('list', planList),
```
Comment (j-selby) [inline cassdegrees/ui/urls.py]: for consistency, should this end with `/`?

File: cassdegrees/ui/views.py [hunk 4]
```diff
@@ -14,12 +14,20 @@ def index(request):
         {'url': "/api/model/degree/", 'img': "../static/img/create_plan_img.png", 'label': "Create Plan"},
         {'url': "/api/model/subplan/", 'img': "../static/img/create_subplan_img.png", 'label': "Create Subplan"},
         {'url': "", 'img': "../static/img/create_list_img.png", 'label': "Create List"},
-        {'url': "", 'img': "../static/img/open_existing_img.png", 'label': "Open Existing"},
+        {'url': "/list", 'img': "../static/img/open_existing_img.png", 'label': "Open Existing"},
```
Comment (j-selby) [inline cassdegrees/ui/views.py]: Ditto with above.

File: cassdegrees/ui/views.py [hunk 5]
```diff
@@ -14,12 +14,20 @@ def index(request):
...
         {'url': "/api/model/course/", 'img': "../static/img/manage_courses_img.png", 'label': "Manage Courses"}
     ]
 
     return render(request, 'index.html', context={'buttons': buttons})
 
+def planList(request):
+    # Generates a table based on whatever JSON object is stored in 'data'
+    # NOTE: data must be an array of dictionaries, with matching labels for each item
```
Comment (j-selby) [inline cassdegrees/ui/views.py]: This should be a doc comment, no? I believe the plan was to use PEP-8, which would describe that.

**Review by j-selby — CHANGES_REQUESTED** [review]
Looking good!

File: cassdegrees/ui/urls.py (hunk 3 above)
Comment (Daniel-Jang) [reply cassdegrees/ui/urls.py]: Yes it should, +1 here

**Comment by Daniel-Jang** [comment]
I'd be happy for you to pull to develop once those minor changes suggested by James has been addressed.

**Review by Daniel-Jang — APPROVED** [review]

File: cassdegrees/ui/views.py [hunk 6]
```diff
@@ -14,12 +14,22 @@ def index(request):
...
 
+def planList(request):
+    """ Generates a table based on whatever JSON object is stored in 'data'
+
+    :param request:
+    :return <class django.http.response.HttpResponse>:
+    """
+    subplans = requests.get(request.build_absolute_uri('/api/model/subplan/?format=json')).json()
+    degree = requests.get(request.build_absolute_uri('/api/model/degree/?format=json')).json()
+    print("TEST", type(render(request, 'list.html', context={'subplans': subplans, 'degrees': degree})))
```
Comment (j-selby) [inline cassdegrees/ui/views.py]: Is this TEST here for a reason?

**Review by j-selby — APPROVED** [review]

**Pull request merged**